tests/unit/restapi/v1/test_experiments.py

```diff
@@ -0,0 +1,859 @@
+            assert isinstance(group["name", str])
+
+    # Validate the EntryPointRef structure
+    for entrypoint in response["entrypoints"]:
+        assert isinstance(entrypoint["id"], int)
+        assert isinstance(entrypoint["url"], str)
+        assert isinstance(entrypoint["name"], str)
+        for group in entrypoint["group"]:
+            assert isinstance(group["id"], int)
+            assert isinstance(group["url"], str)
+            assert isinstance(group["name", str])
+
+    # Validate the TagRef structure
+    for tag in response["tags"]:
+        assert isinstance(tag["id"], int)
+        assert isinstance(tag["name"], str)
+        assert isinstance(tag["url"], str)
+
+
+def assert_retrieving_experiment_by_id_works(
+    client: FlaskClient,
+    experiment_id: int,
+    expected: dict[str, Any],
+) -> None:
+    """Assert that retrieving an experiment by id works.
+
+    Args:
+        client: The Flask test client.
+        experiment_id: The id of the experiment to retrieve.
+        expected: The expected response from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_EXPERIMENTS_ROUTE}/{experiment_id}", follow_redirects=True
+    )
+    assert response.status_code == 200 and response.get_json() == expected
+
+
+def assert_retrieving_experiment_entrypoints_works(
+    client: FlaskClient,
+    experiment_id: int,
+    expected: dict[str, Any],
+) -> None:
+    """Assert that retrieving an experiment's associated entrypoints can be retrieved
+    from an experiment with the specified id.
+
+    Args:
+        client: The Flask test client.
+        experiment_id: The id of the experiment to retrieve entrypoints.
+        expected_entrypoints: The expected entrypoints from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_EXPERIMENTS_ROUTE}/{experiment_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 200 and response == expected
+
+
+def assert_adding_experiment_entrypoints_works(
+    client: FlaskClient,
+    experiment_id: int,
+    entrypoints: list[int],
+    expected: dict[str, Any],
+) -> None:
+    """Assert that an experiment's associated entrypoints can be updated for an
+    experiment with specified id.
+
+    Args:
+        client: The Flask test client.
+        experiment_id: The id of the experiment to add entrypoints.
+        expected: The expected response from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+    payload = {"entrypoints": entrypoints}
+    response = client.post(
```

Comment:
adding entrypoints is a PUT and the payload needs to be the complete experiment object.